File: 0081 search-in-rotated-sorted-array-ii/solution.java
class Solution {
    public boolean search(int[] nums, int target) {
     return mbin(nums,target,0,nums.length-1);   
    }

    public boolean mbin(int arr[],int target,int left,int right){
        if(left>right){
            return false;
        }
        int mid = left + (right - left) / 2;
        if(arr[mid]==target){
            return true;
        }
        if (arr[left] == arr[mid] && arr[mid] == arr[right]) {
            return mbin(arr, target, left + 1, right - 1);
        }
        if(arr[mid] >= arr[left]){
            if(arr[left]<=target && target<=arr[mid]){
                return mbin(arr,target,left,mid-1);
            }
            else{
                return mbin(arr,target,mid+1,right);
            }
        }
        else{
            if(arr[mid]<=target && target<=arr[right]){
                return mbin(arr,target,mid+1,right);
            }
            else{
                return mbin(arr,target,left,mid-1);
            }
        }
    }
}

File: 0081 search-in-rotated-sorted-array-ii/solution.rs
impl Solution {
    pub fn search(nums: Vec<i32>, target: i32) -> bool {
        let mut l: isize = 0;
        let mut r: isize = nums.len() as isize - 1;
        while l <= r {
            let mid = l + (r - l) / 2;
            let mid_val = nums[mid as usize];
            if mid_val == target {
                return true;
            }
            if nums[l as usize] == mid_val && mid_val == nums[r as usize] {
                l += 1;
                r -= 1;
            } else if mid_val >= nums[l as usize] {
                if target >= nums[l as usize] && target < mid_val {
                    r = mid - 1;
                } else {
                    l = mid + 1;
                }
            } else {
                if target > mid_val && target <= nums[r as usize] {
                    l = mid + 1;
                } else {
                    r = mid - 1;
                }
            }
        }
        false
    }
}
